File: Golang/Product of Array Except Self.go
func productExceptSelf(nums []int) []int {
    res := make([]int, len(nums))
    res[0] = 1
    for i:=1;i < len(nums);i++ {
        res[i] = res[i-1] * nums[i-1]
    }
    temp := 1
    for i:=len(nums)-1;i > -1;i-- {
        res[i] = res[i] * temp
        temp = temp * nums[i]
    }
    return res
}

File: Python3/Product of Array Except Self.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        res = [1] * len(nums)
        res[0] = 1
        for i in range(1, len(nums)):
            res[i] = nums[i-1] * res[i-1]
        temp = 1
        for i in range(len(nums)-1, -1, -1):
            res[i] = res[i] * temp
            temp = temp * nums[i]

        return res
        